x-pack/plugins/endpoint/server/routes/index_pattern.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter, Logger, RequestHandler } from 'kibana/server';
+import { EndpointAppContext } from '../types';
+import { IndexPatternGetParamsResult } from '../../common/types';
+import { indexPatternGetParamsSchema } from '../../common/schema/index_pattern';
+import { IndexPatternRetriever } from '../index_pattern';
+
+function handleIndexPattern(
+  log: Logger,
+  indexRetriever: IndexPatternRetriever
+): RequestHandler<IndexPatternGetParamsResult> {
+  return async (context, req, res) => {
+    try {
+      return res.ok({
+        body: {
+          indexPattern: await indexRetriever.get(
+            context.core.savedObjects.client,
+            req.params.datasetPath
+          ),
+        },
+      });
+    } catch (error) {
+      log.warn(error);
+      return res.notFound({ body: error });
+    }
+  };
+}
+
+export function registerIndexPatternRoute(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const log = endpointAppContext.logFactory.get('index_pattern');
+
+  router.get(
+    {
+      path: '/api/endpoint/index_pattern/{datasetPath}',
```

Comment:
we may want to externalize this, but it is your call.

Reply:
@nnamdifrankie could you expand on what you mean by externalize? This handler is so that the endpoint's frontend can retrieve the index pattern for alerts which is currently the same as events.

Reply:
Sorry I was not too clear. There is already a constant and it is externalized https://github.com/elastic/kibana/blob/8489efc034dddfae18913830f50091eb603100ff/x-pack/plugins/endpoint/common/types.ts#L32. And you may also want to do https://github.com/elastic/kibana/blob/master/x-pack/plugins/endpoint/server/routes/alerts/index.ts#L13 for own local base if you want.

Reply:
Oh got it. Thanks, I'll update 👍